src/storage/rdb.cc

```diff
@@ -1005,3 +1000,35 @@ Status RDB::rdbSaveBinaryDoubleValue(double val) {
   memrev64ifbe(&val);
   return stream_->Write((const char *)(&val), sizeof(val));
 }
+
+Status RDB::rdbSaveZipListObject(const std::string &elem) {
+  // calc total ziplist size
+  uint prevlen = 0;
+  const size_t ziplist_size = zlHeaderSize + zlEndSize + elem.length() +
+                              ZipList::ZipStorePrevEntryLength(nullptr, prevlen) +
+                              ZipList::ZipStoreEntryEncoding(nullptr, elem.length());
+  auto zl_string = std::string(ziplist_size, 0);
+  auto zl_ptr = reinterpret_cast<unsigned char *>(&zl_string[0]);
+
+  // set ziplist header
+  ZipList::SetZipListBytes(zl_ptr, intrev32ifbe(ziplist_size));
```

Comment:
`ziplist_size` is not a uint32, right? Would we check and cast it to u32?

Reply:
done